lib/auth/userpreferences/userpreferencesv1/service_test.go

```diff
@@ -0,0 +1,262 @@
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		userName string
+		req      *userpreferencesv1.GetUserPreferencesRequest
+		want     *userpreferencesv1.GetUserPreferencesResponse
+		wantErr  assert.ErrorAssertionFunc
+	}{
+		{
+			name:     "success",
+			userName: defaultUser,
+			req: &userpreferencesv1.GetUserPreferencesRequest{
+				Username: defaultUser,
+			},
+			want: &userpreferencesv1.GetUserPreferencesResponse{
+				Preferences: &userpreferencesv1.UserPreferences{
+					Assist: &userpreferencesv1.AssistUserPreferences{
+						PreferredLogins: []string{},
+						ViewMode:        userpreferencesv1.AssistViewMode_ASSIST_VIEW_MODE_DOCKED,
+					},
+					Theme: userpreferencesv1.Theme_THEME_LIGHT,
+					Onboard: &userpreferencesv1.OnboardUserPreferences{
+						PreferredResources: []userpreferencesv1.Resource{},
+					},
+				},
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name:     "access denied - incorrect user",
+			userName: defaultUser,
+			req: &userpreferencesv1.GetUserPreferencesRequest{
+				Username: "non-existent-user",
+			},
+			want:    nil,
+			wantErr: assert.Error,
+		},
+		{
+			name:     "access denied - RBAC rule",
+			userName: noAccessUser,
+			req: &userpreferencesv1.GetUserPreferencesRequest{
+				Username: "test-user-no-access",
+			},
+			want:    nil,
+			wantErr: assert.Error,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctxs, svc := initSvc(t)
+
+			got, err := svc.GetUserPreferences(ctxs[tt.userName], tt.req)
+			tt.wantErr(t, err)
+
+			if tt.want != nil {
+				require.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func TestService_UpsertUserPreferences(t *testing.T) {
+	t.Parallel()
+
+	defaultPreferences := &userpreferencesv1.UserPreferences{
+		Assist: &userpreferencesv1.AssistUserPreferences{
+			PreferredLogins: []string{},
+			ViewMode:        userpreferencesv1.AssistViewMode_ASSIST_VIEW_MODE_DOCKED,
+		},
+		Theme: userpreferencesv1.Theme_THEME_LIGHT,
+		Onboard: &userpreferencesv1.OnboardUserPreferences{
+			PreferredResources: []userpreferencesv1.Resource{},
+		},
+	}
+
+	tests := []struct {
+		name     string
+		userName string
+		req      *userpreferencesv1.UpsertUserPreferencesRequest
+		wantErr  assert.ErrorAssertionFunc
+	}{
+		{
+			name:     "success",
+			userName: defaultUser,
+			req: &userpreferencesv1.UpsertUserPreferencesRequest{
+				Username:    defaultUser,
+				Preferences: defaultPreferences,
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name:     "access denied - incorrect user",
+			userName: defaultUser,
+			req: &userpreferencesv1.UpsertUserPreferencesRequest{
+				Username:    "non-existent-user",
+				Preferences: defaultPreferences,
+			},
+			wantErr: assert.Error,
+		},
+		{
+			name:     "access denied - RBAC rule",
+			userName: noAccessUser,
+			req: &userpreferencesv1.UpsertUserPreferencesRequest{
+				Username:    noAccessUser,
+				Preferences: defaultPreferences,
+			},
+			wantErr: assert.Error,
+		},
```

Comment:
Shouldn't a user be able to update their own preferences?

```suggestion
		{
			name:     "success",
			userName: noAccessUser,
			req: &userpreferencesv1.UpsertUserPreferencesRequest{
				Username:    noAccessUser,
				Preferences: defaultPreferences,
			},
			wantErr: assert.NoError,
		},
```

Reply:
In that case, we should probably remove RBAC for user preferences. Just let you do whatever you want with your own preferences. I followed the same rules as the other related resources (password for example).